app/src/main/java/com/fpvout/digiview/helpers/StreamDumper.java

```diff
@@ -0,0 +1,71 @@
+package com.fpvout.digiview.helpers;
+
+import android.content.Context;
+import android.os.Environment;
+import android.os.Handler;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+
+public class StreamDumper  {
+
+    private FileOutputStream fos;
+    private boolean bytesWritten = false;
+
+    private final File dumpDir;
+    private File streamDump;
+    private File streamAmbient;
+    private File outFile;
+    private final Context context;
+
+    public StreamDumper(Context context, String defaultPath){
+        this.context = context;
+        dumpDir = new File(defaultPath);
+
+        dumpDir.mkdirs();
+    }
+
+    public void dump(byte[] buffer, int offset, int receivedBytes) {
+
+        try {
+            fos.write(buffer, offset, receivedBytes);
+            bytesWritten = true;
+        } catch (IOException exception) {
+            exception.printStackTrace();
+        }
+    }
+
+    public void init(String videoFileName, String ambientAudioFileName, String outFileFileName) {
+        try {
+            streamDump = new File(dumpDir, videoFileName);
+            streamAmbient = new File(dumpDir, ambientAudioFileName);
+            outFile = new File(dumpDir, outFileFileName);
+            fos = new FileOutputStream(streamDump);
+            bytesWritten = false;
+        } catch (IOException exception) {
+            exception.printStackTrace();
+        }
+    }
+
+    public void stop(Handler completeHandler) {
+        try {
+            if(fos != null){
+                fos.flush();
+                fos.close();
+
+                if(bytesWritten) {
+                    new Mp4Muxer(this.context, dumpDir, streamDump, streamAmbient,outFile).start();
```

Comment:
Would be neat to do this real-time ! (don't know if it's actually easy / doable)
Probably not a total requirement at the moment, but it would means that if app stops during recording you would loose everything here... (well you still have a raw h264 stream and a .aac record I guess 😛 )

Reply:
yes was thinking about that. Maybe on fallback it searches for unfinished h264/aac and merges them.

Reply:
a bit like repair mode on gopro ? I like it ^^

Reply:
added this feature